src/XRP/xrp-payment.ts

```diff
@@ -0,0 +1,79 @@
+import { Payment } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import XRPCurrencyAmount from './xrp-currency-amount'
+import XRPPath from './xrp-path'
+
+// TODO(amiecorso): Modify this object to use X-Address format.
+export default class XRPPayment {
+  public static from(payment: Payment): XRPPayment | undefined {
+    let amount
+    const paymentAmountValue = payment.getAmount()?.getValue()
+    if (paymentAmountValue) {
+      amount = XRPCurrencyAmount.from(paymentAmountValue)
```

Comment:
What is `XRPCurrencyAmount.from(undefined)`? Would that let us collapse some of this conditional stuff?

Then this could look like:
```ts
    const paymentAmountValue = payment.getAmount()?.getValue()
    const amount = XRPCurrencyAmount.from(paymentAmountValue)
    if (!amount) {
      return undefined // amount is required
    }
```

Reply:
I think the hiccup with this approach is that `paymentAmountValue` potentially has a value of `undefined`, based on how the protobufs are defined... so in order to feed it to `XRPCurrencyAmount.from(paymentAmountValue)`, the signature of `.from(...)` would have to include `undefined` as a type option for its parameter.   Do we think it's better to give all the `.from()` methods the option to take an `undefined` parameter and then simply immediately return `undefined`?

Reply:
@keefertaylor ☝️

Reply:
Like my comment in https://github.com/xpring-eng/Xpring-JS/pull/249#discussion_r395053932, I think you could do a:
```

Reply:
Like my comment in https://github.com/xpring-eng/Xpring-JS/pull/249#discussion_r395053932

I think you could do a:
```js
const amount = paymentAmountValue && XRPCurrencyAmount.from(paymentAmountValue)
```

I do see you have an additional check on if the `XRPCurrencyAmount.from()` fails, even with a truthy `paymentAmountValue`, so you could move that:
```
if (!amount) return undefined
```
to the outside.